### PR TITLE
app-misc/recoll: Version bump to 1.40.1

### PR DESCRIPTION
**Pull request by ppurka**
Bug: https://bugs.gentoo.org/937911
Signed-off-by: Punarbasu Purkayastha <ppurka@gmail.com>

- Update to recoll-1.40.1
- Update to meson build instead of autoconf
- Add support for qt6. The ebuild favours qt6 over qt5 if both qt5 and qt6 USE flags are enabled.
- Update bug report URL in metadata (reported by `pkgcheck`)
- Remove HTML doc installation  - it was creating trouble

---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

**Comment by ppurka** [comment]
Diff between the existing version in portage and the new version (quite a bit of diff):
```diff
--- recoll-1.37.4-r1.ebuild     2024-09-09 20:09:53.945024472 +0100
+++ recoll-1.40.1.ebuild        2024-09-09 20:40:06.990915850 +0100
@@ -5,20 +5,20 @@
 
 PYTHON_COMPAT=( python3_{10..12} )
 
-inherit linux-info optfeature python-single-r1 qmake-utils systemd
+inherit linux-info optfeature python-single-r1 qmake-utils meson systemd
 
 DESCRIPTION="Personal full text search package"
-HOMEPAGE="https://www.lesbonscomptes.com/recoll/"
-SRC_URI="https://www.lesbonscomptes.com/recoll/${P}.tar.gz"
+HOMEPAGE="https://www.recoll.org"
+SRC_URI="https://www.recoll.org/${P}.tar.gz"
 
 LICENSE="GPL-2+"
 SLOT="0"
-KEYWORDS="amd64 ~arm64 x86"
+KEYWORDS="~amd64 ~arm64 ~x86"
 
-IUSE="camelcase chm +inotify qt5 session +spell systemd webengine"
+IUSE="camelcase chm +inotify qt5 qt6 session +spell systemd webengine"
 REQUIRED_USE="
        session? ( inotify )
-       webengine? ( qt5 )
+       webengine? ( || ( qt5 qt6 ) )
        ${PYTHON_REQUIRED_USE}
 "
 
@@ -32,13 +32,19 @@
                dev-libs/chmlib
                dev-python/pychm
        )
-       qt5? (
-               dev-qt/qtcore:5
-               dev-qt/qtgui:5
-               dev-qt/qtnetwork:5
-               dev-qt/qtprintsupport:5
-               dev-qt/qtwidgets:5
-               webengine? ( dev-qt/qtwebengine:5[widgets] )
+       qt6? (
+               dev-qt/qtbase:6[gui,network,widgets]
+               webengine? ( dev-qt/qtwebengine:6[widgets] )
+       )
+       !qt6? (
+               qt5? (
+                       dev-qt/qtcore:5
+                       dev-qt/qtgui:5
+                       dev-qt/qtnetwork:5
+                       dev-qt/qtprintsupport:5
+                       dev-qt/qtwidgets:5
+                       webengine? ( dev-qt/qtwebengine:5[widgets] )
+               )
        )
        session? (
                inotify? (
@@ -53,7 +59,10 @@
 "
 
 BDEPEND="
-       qt5? ( dev-qt/linguist-tools:5 )
+       qt6? ( dev-qt/qttools:6 )
+       !qt6? (
+               qt5? ( dev-qt/linguist-tools:5 )
+       )
 "
 
 RDEPEND="
@@ -75,37 +84,46 @@
 }
 
 src_configure() {
-       use qt5 && export QMAKE="$(qt5_get_bindir)/qmake"
+       # qt6 has preference over qt5
+       if use qt6; then
+               export QMAKE="$(qt6_get_bindir)/qmake"
+       elif use qt5; then
+               export QMAKE="$(qt5_get_bindir)/qmake"
+       fi
 
-       local myeconfargs=(
-               $(use_enable camelcase)
-               $(use_enable chm python-chm)
-               $(use_enable session x11mon)
-               $(use_enable qt5 qtgui)
-               $(use_enable spell python-aspell)
-               $(use_enable webengine)
-               $(use_with inotify)
-               $(use_with spell aspell)
-               $(use_with systemd)
-               --with-system-unit-dir="$(systemd_get_systemunitdir)"
-               --with-user-unit-dir="$(systemd_get_userunitdir)"
-               --disable-webkit
-               --without-fam
-               --enable-recollq
+       local emesonargs=(
+               $(meson_use camelcase)
+               $(meson_use chm python-chm)
+               $(meson_use inotify)
+               $(meson_use session x11mon)
+               $(meson_use spell aspell)
+               $(meson_use spell python-aspell)
+               $(meson_use systemd)
+               $(meson_use webengine)
+               -Dfam=false
+               -Drecollq=true
+               -Dwebkit=false
+               -Dsystemd-system-unit-dir="$(systemd_get_systemunitdir)"
+               -Dsystemd-user-unit-dir="$(systemd_get_userunitdir)"
        )
-       use qt5 && myeconfargs+=( $(usex webengine "--disable-webpreview" "--enable-webpreview" ) )
 
-       econf "${myeconfargs[@]}"
+       if use qt5 || use qt6; then
+               emesonargs+=( -Dqtgui=true )
+               emesonargs+=( $(usex webengine "-Dwebpreview=true" "-Dwebpreview=false") )
+       else
+               emesonargs+=( -Dqtgui=false )
+       fi
+
+       meson_src_configure
 }
 
 src_install() {
-       emake STRIP="$(type -P true || die)" DESTDIR="${D}" install
+       meson_install --strip
        rm -rf "${D}/$(python_get_sitedir)"/*.egg-info || die
        python_optimize
 
        # html docs should be placed in /usr/share/doc/${PN}/html
-       dodoc -r "${ED}"/usr/share/recoll/doc/.
-       rm -r "${ED}/usr/share/recoll/doc" || die
+       rm -r "${ED}/usr/share/${PN}/doc" || die
        find "${D}" -name '*.la' -delete || die
 }
```

**Review by a17r — CHANGES_REQUESTED** [review]
> Add support for qt6. The ebuild favours qt6 over qt5 if both qt5 and qt6 USE flags are enabled.

Please consider dropping Qt5 right away.

**Comment by ppurka** [comment]
Removed qt5 from the ebuild. New diff between existing recoll and the new recoll ebuilds:
```diff
--- recoll-1.37.4-r1.ebuild     2024-09-09 20:09:53.945024472 +0100
+++ recoll-1.40.1.ebuild        2024-09-10 19:30:39.305989189 +0100
@@ -5,20 +5,20 @@
 
 PYTHON_COMPAT=( python3_{10..12} )
 
-inherit linux-info optfeature python-single-r1 qmake-utils systemd
+inherit linux-info optfeature python-single-r1 qmake-utils meson systemd
 
 DESCRIPTION="Personal full text search package"
-HOMEPAGE="https://www.lesbonscomptes.com/recoll/"
-SRC_URI="https://www.lesbonscomptes.com/recoll/${P}.tar.gz"
+HOMEPAGE="https://www.recoll.org"
+SRC_URI="https://www.recoll.org/${P}.tar.gz"
 
 LICENSE="GPL-2+"
 SLOT="0"
-KEYWORDS="amd64 ~arm64 x86"
+KEYWORDS="~amd64 ~arm64 ~x86"
 
-IUSE="camelcase chm +inotify qt5 session +spell systemd webengine"
+IUSE="camelcase chm +inotify qt6 session +spell systemd webengine"
 REQUIRED_USE="
        session? ( inotify )
-       webengine? ( qt5 )
+       webengine? ( qt6 )
        ${PYTHON_REQUIRED_USE}
 "
 
@@ -32,13 +32,9 @@
                dev-libs/chmlib
                dev-python/pychm
        )
-       qt5? (
-               dev-qt/qtcore:5
-               dev-qt/qtgui:5
-               dev-qt/qtnetwork:5
-               dev-qt/qtprintsupport:5
-               dev-qt/qtwidgets:5
-               webengine? ( dev-qt/qtwebengine:5[widgets] )
+       qt6? (
+               dev-qt/qtbase:6[gui,network,widgets]
+               webengine? ( dev-qt/qtwebengine:6[widgets] )
        )
        session? (
                inotify? (
@@ -53,7 +49,7 @@
 "
 
 BDEPEND="
-       qt5? ( dev-qt/linguist-tools:5 )
+       qt6? ( dev-qt/qttools:6 )
 "
 
 RDEPEND="
@@ -75,37 +71,37 @@
 }
 
 src_configure() {
-       use qt5 && export QMAKE="$(qt5_get_bindir)/qmake"
+       use qt6 && export QMAKE="$(qt6_get_bindir)/qmake"
 
-       local myeconfargs=(
-               $(use_enable camelcase)
-               $(use_enable chm python-chm)
-               $(use_enable session x11mon)
-               $(use_enable qt5 qtgui)
-               $(use_enable spell python-aspell)
-               $(use_enable webengine)
-               $(use_with inotify)
-               $(use_with spell aspell)
-               $(use_with systemd)
-               --with-system-unit-dir="$(systemd_get_systemunitdir)"
-               --with-user-unit-dir="$(systemd_get_userunitdir)"
-               --disable-webkit
-               --without-fam
-               --enable-recollq
+       local emesonargs=(
+               $(meson_use camelcase)
+               $(meson_use chm python-chm)
+               $(meson_use inotify)
+               $(meson_use qt6 qtgui)
+               $(meson_use session x11mon)
+               $(meson_use spell aspell)
+               $(meson_use spell python-aspell)
+               $(meson_use systemd)
+               $(meson_use webengine)
+               -Dfam=false
+               -Drecollq=true
+               -Dwebkit=false
+               -Dsystemd-system-unit-dir="$(systemd_get_systemunitdir)"
+               -Dsystemd-user-unit-dir="$(systemd_get_userunitdir)"
        )
-       use qt5 && myeconfargs+=( $(usex webengine "--disable-webpreview" "--enable-webpreview" ) )
 
-       econf "${myeconfargs[@]}"
+       use qt6 && emesonargs+=( $(usex webengine "-Dwebpreview=true" "-Dwebpreview=false") )
+
+       meson_src_configure
 }
 
 src_install() {
-       emake STRIP="$(type -P true || die)" DESTDIR="${D}" install
+       meson_install --strip
        rm -rf "${D}/$(python_get_sitedir)"/*.egg-info || die
        python_optimize
 
        # html docs should be placed in /usr/share/doc/${PN}/html
-       dodoc -r "${ED}"/usr/share/recoll/doc/.
-       rm -r "${ED}/usr/share/recoll/doc" || die
+       rm -r "${ED}/usr/share/${PN}/doc" || die
        find "${D}" -name '*.la' -delete || die
 }
```

File: app-misc/recoll/recoll-1.40.1.ebuild [hunk 1]
```diff
@@ -60,9 +50,6 @@ DEPEND="
 
 BDEPEND="
 	qt6? ( dev-qt/qttools:6 )
-	!qt6? (
-		qt5? ( dev-qt/linguist-tools:5 )
-	)
```
Comment (a17r) [inline app-misc/recoll/recoll-1.40.1.ebuild]: You will likely want to have `qttools[linguist]` if this was going to fulfill the purpose that `dev-qt/linguist-tools:5` had.

**Review by a17r — CHANGES_REQUESTED** [review]

**Comment by ppurka** [comment]
Updated the ebuild. Thanks for the advice!

**Pull request closed without merging**